tracks/elm/exercises/twelve-days/mentoring.md

```diff
@@ -0,0 +1,41 @@
+
+### Reasonable solutions
+
+Simply listing out the complete text of all verses is the simplest and easiest solution, and would be a reasonable initial choice in real life. Then, if changes were required to the song, it would make sense to refactor to remove the duplication, and then make the changes. However this solution doesn't really help the student learn elm, so if they have done this it makes sense to encourage them to make a further iteration.
+
+All solutions need a way of turning an `Int` in to a string ordinal, and in to a string describing the gift for a day. `Dict.fromList`, `Array.fromList`, `List` or `case statements` are the normal ways to do this, and there is not much to choose between them.
+
+It is generally best to iterate over the verses using `List.map` or similar. It is possible to use recursion, but it makes naming harder, and is more complex than it needs to be.
+
+It is possible to use a type (like the one below) to contain all the information for a day (the ordinal and the gift for the day). This keeps the ordinal and gift strings closer together, which is good, but adds some noise, which is bad. Either is acceptable.
+
+```elm
+type alias NthDay =
+    { nthDay : Int
+	, ordinal : String
+    , gift : String
+    }
+```
+
+["Ceddlyburge" has a example solution](https://exercism.io/tracks/elm/exercises/twelve-days/solutions/7b614615e44f4ea6b6a0aa1b919054de) that uses case statements.
+	
+### Common suggestions
+
+Naming is difficult in this solution, and most students do not choose good names. Generally the name should be one level of abstraction up from what the code does (so it still means something, but you can change the implementation without having to change the name). `onTheNthDay` and `whatWasGiven` are reasonable choices for commonly used functions. `onTheNthDay` doesn't mention Christmas, so would allow the code to be changed for a different event ('on the first day of easter ...') and still make sense, while still being meaningful enough to be useful. 'onTheNth' is also possible, which would allow the code to change to 'on the second year of the new millenium' without having to change the name, and again the name is still meaningful enough to be useful.
+
+A lot of students stuggle to concatenate the verses without a lot of special case code and conditionals. Most of this can be avoided by adding `,` and `, and` to the gift strings.
+
+### Talking points
+
+In the real world, the entry function would probably return a `Maybe` or a `Result` to indicate that it can fail / can recieve invalid input. However, this function already defined by the exercise without this (I think to keep the exercise relatively simple). This means that we have to make choices about what to do with out of bounds inputs. Using `clamp` or similar is one possibility, to transform out of range inputs in to in range ones. The other options are to return non sensical output, or explicit error output. 
```

Comment:
```suggestion
In the real world, the entry function would probably return a `Maybe` or a `Result` to indicate that it can fail / can recieve invalid input. However, this function already defined by the exercise without this (I think to keep the exercise relatively simple). This means that we have to make choices about what to do with out of bounds inputs. Using `clamp` or similar is one possibility, to transform out of range inputs in to in range ones. The other options are to return nonsensical output, or explicit error output. 
```